src/gui/toolbarMenubar/ToolPageSpinner.cpp

```diff
@@ -35,24 +62,52 @@ auto ToolPageSpinner::getNewToolIcon() -> GtkWidget* {
 }
 
 auto ToolPageSpinner::newItem() -> GtkToolItem* {
-    GtkToolItem* it = gtk_tool_item_new();
-
-    GtkWidget* hbox = gtk_box_new(GTK_ORIENTATION_HORIZONTAL, 1);
+    if (this->pageSpinner->hasWidget()) {
+        this->pageSpinner->removeWidget();
+    }
+    GtkWidget* spinner = gtk_spin_button_new_with_range(0, 1, 1);
+    this->pageSpinner->setWidget(spinner);
+    gtk_orientable_set_orientation(reinterpret_cast<GtkOrientable*>(spinner), orientation);
 
     GtkWidget* pageLabel = gtk_label_new(_("Page"));
-    GtkWidget* spinner = this->pageSpinner->getWidget();
+    this->lbPageNo = gtk_label_new("");
```

Comment:
This will leak memory when `newItem` is called again.

Reply:
If i understand that correctly we would need a lot more of these gobj unrefs. For example in line 75 for that box = gtk_box_new(orientation, 1); 
Is that right?

Reply:
Yeah, the important thing is to understand that [gtk objects are reference counted](https://developer.gnome.org/programming-guidelines/stable/memory-management.html.en). You need an unref if you are going to "lose" the reference, or a ref if you're going to store an additional one. For example:
```
GtkWidget* box = gtk_new_new(orientation, 1); // this has a reference count of 1
gtk_widget_add(someContainer, box); // someContainer will hold a reference of box and increase the reference count to 2

// we don't need box anymore, so decrease the reference count. Otherwise we'll get a memory leak later!
g_object_unref(box); // box ref count 2 --> 1

// when the container is destroyed, ref count of box drops from 1 --> 0 and box is deallocated
```

Second example:
```
GtkLabel* label = gtk_label_new(); // ref count of 1
this->label = label; // here we "transfer" the reference to the class field
// we don't need to do a unref here because it should be unref'd at some other point (e.g. destructor)
```

Reply:
Ok, thanks for that explanation. I'm gonna try to fix that.